components/automate-ui/src/app/components/projects-dropdown/projects-dropdown.component.html

```diff
@@ -14,15 +14,34 @@
   </button>
   <chef-click-outside (clickOutside)="closeDropdown()">
     <chef-dropdown [attr.visible]="active">
-      <chef-checkbox
-        *ngFor="let project of projectsArray"
-        [checked]="project.checked"
-        [attr.title]="project.name"
-        (change)="projectChecked($event.detail, project)"
-        (keydown.enter)="closeDropdown()"
-        (keydown.esc)="closeDropdown()"
-        (keydown.arrowup)="moveFocus($event)"
-        (keydown.arrowdown)="moveFocus($event)">{{ project['name'] }}</chef-checkbox>
+
+      <div id="filter-container">
+        <input 
+          chefInput
+          type="text"
+          [(ngModel)]="filterValue"
+          placeholder="Filter projects..." 
+          (keyup)="handleFilterKeyUp()"
+          />
+      </div>
+
+      <div id="project-container">
+        <chef-checkbox
+          *ngFor="let project of filteredProjects"
+          [checked]="project.checked"
+          [attr.title]="project.name"
+          (change)="projectChecked($event.detail, project)"
+          (keydown.enter)="closeDropdown()"
```

Comment:
in other places I've seen we use a pattern of passing any keyboard event to a handler function, then which decides how to react depending on the key. I do like that we can target specific key events here, making it really explicit what each action will trigger. I wonder if it's faster either way?

Reply:
I opted to not touch the functionality that was already in place, but I think it could be nice to take in any keypress, and then I wonder if we could even map it out in a reusable function so that we have consistent accessibility across the site.

Reply:
oo that sounds great! like a directive you could put on components maybe?